doc/hardware/porting/board_porting.rst

```diff
@@ -133,57 +171,127 @@ board directory to a separate repository once it's working.)
 
   The board directory name does not need to match the name of the board.
   Multiple boards can even defined be in one directory.
-  For example, for boards with multi-core SoC, a logical board might be created
-  for each core following the naming scheme `<board>_<soc-core>`, with definitions
-  for all of these different boards defined inside the same directory. This and
-  similar schemes are common for upstream vendor boards.
 
 Your board directory should look like this:
 
 .. code-block:: none
 
    boards/<ARCH>/plank
+   ├── board.yml
    ├── board.cmake
    ├── CMakeLists.txt
    ├── doc
    │   ├── plank.png
    │   └── index.rst
-   ├── Kconfig.board
+   ├── Kconfig.plank
    ├── Kconfig.defconfig
    ├── plank_defconfig
+   ├── plank_<identifier>_defconfig
    ├── plank.dts
+   ├── plank_<identifier>.dts
+   ├── plank_<identifier>.overlay
    └── plank.yaml
 
 Replace ``plank`` with your board's name, of course.
 
 The mandatory files are:
 
-#. :file:`plank.dts`: a hardware description in :ref:`devicetree
-   <dt-guide>` format. This declares your SoC, connectors, and any
-   other hardware components such as LEDs, buttons, sensors, or communication
-   peripherals (USB, BLE controller, etc).
+#. :file:`board.yml`: a YAML file describe the highlevel meta data of the board
+   such as the board name, SoCs, and variants.
+   Cpusets for multi-core SoCs are not described in this file as they are
+   inherited from the SoC's YAML description.
+
+#. :file:`plank.dts` or :file:`plank_<identifier>.dts`: a hardware description
+   in :ref:`devicetree <dt-guide>` format. This declares your SoC, connectors,
+   and any other hardware components such as LEDs, buttons, sensors, or
+   communication peripherals (USB, BLE controller, etc).
+
+#. :file:`Kconfig.plank`: the base software configuration for selecting SoC and
+   other board and SoC related setting. No Kconfig settings outside the board
+   and SoC tree must be selected. To select general Zephyr Kconfig settings the
```

Comment:
What is intended is to have a re-usable Kconfig tree for the hardware parts, board, soc, and arch. which can then be sourced into other trees. One example being sysbuild.

This would allow for creating default images based on board and SoC.
Anther use-case which has emerged is the possibility of an early / limited Kconfig run, to just handle the board and soc parts.

For this to work properly, the settings in the mentioned Kconfig.plank cannot select a Zephyr driver or compile option, but it can select a SoC.

Hope this make the intention clearer, and i'd be happy for any proposals which can clarify above text without writing a novel.

Reply:
> What is intended is to have a re-usable Kconfig tree

Should this be given a name, and described somewhere, and have from here a link to that?

Reply:
> > What is intended is to have a re-usable Kconfig tree
> 
> Should this be given a name, and described somewhere, and have from here a link to that?

Good suggestion.

Added an entry here: https://github.com/zephyrproject-rtos/zephyr/issues/51831

The Kconfig hardware re-usable tree is something new, and doesn't belong in board / soc porting but deserve an independent decsription / pags (where ?!).
Thus outside scope of this PR.